src/Microsoft.SqlTools.ServiceLayer/Utility/TaskExtensions.cs

```diff
@@ -18,33 +18,83 @@ public static class TaskExtensions
                 {
                     return;
                 }
                 
-                // Construct an error message for an aggregate exception and log it
-                StringBuilder sb = new StringBuilder("Unhandled exception(s) in async task:");
-                foreach (Exception e in task.Exception.InnerExceptions)
+                LogTaskExceptions(task.Exception);
+
+                // Run the continuation task that was provided
+                try
+                {
+                    continuationAction?.Invoke(task);
+                }
+                catch (Exception e)
                 {
-                    sb.AppendLine($"{e.GetType().Name}: {e.Message}");
-                    sb.AppendLine(e.StackTrace);
+                    Logger.Write(LogLevel.Error, $"Exception in exception handling continuation: {e}");
+                    Logger.Write(LogLevel.Error, e.StackTrace);
                 }
-                Logger.Write(LogLevel.Error, sb.ToString());
+            });
+        }
+
+        /// <summary>
+        /// Adds handling to check the Exception field of a task and log it if the task faulted.
+        /// This version allows for async code to be ran in the continuation function.
+        /// </summary>
+        /// <remarks>
+        /// This will effectively swallow exceptions in the task chain. 
+        /// </remarks>
+        /// <param name="antecedent">The task to continue</param>
+        /// <param name="continuationFunc">
+        /// An optional operation to perform after exception handling has occurred
+        /// </param>
+        /// <returns>Task with exception handling on continuation</returns>
+        public static Task ContinueWithOnFaulted(this Task antecedent, Func<Task, Task> continuationFunc)
+        {
+            return antecedent.ContinueWith(task =>
+            {
+                // If the task hasn't faulted or doesn't have an exception, skip processing
+                if (!task.IsFaulted || task.Exception == null)
+                {
+                    return;
+                }
+
+                LogTaskExceptions(task.Exception);
 
                 // Run the continuation task that was provided
-                continuationAction?.Invoke(t);
+                try
+                {
+                    continuationFunc?.Invoke(antecedent).Wait();
+                }
+                catch (Exception e)
+                {
+                    Logger.Write(LogLevel.Error, $"Exception in exception handling continuation: {e}");
+                    Logger.Write(LogLevel.Error, e.StackTrace);
+                }
             });
         }
+
+        private static void LogTaskExceptions(AggregateException exception)
+        {
+            // Construct an error message for an aggregate exception and log it
+            StringBuilder sb = new StringBuilder("Unhandled exception(s) in async task:");
+            foreach (Exception e in exception.InnerExceptions)
```

Comment:
You should flatten the exceptions first. This removes nested aggregates which is quite common